src/java.cpp

```diff
@@ -14,7 +14,8 @@
 
 #define DYNAMIC_PROXY_JS_ERROR -4
 
-long v8ThreadId;
+long v8ThreadIdWIN32;
```

Comment:
Good catch, and great solution, although I don't believe xlc++ can handle the edit. I get the following error on compilation:

invalid operands to binary expression 
('threadId' (aka 'pthread_t') and 'threadId')                         
  if((isWIN32 && (myThreadId == v8ThreadId)) || (!isWIN32 && pthread_equal(myThreadId, v8ThreadId))) {

This is being thrown on:
myThreadId == v8ThreadId 
even though it should have short circuited on the isWIN32 check